usage.go

```diff
@@ -0,0 +1,251 @@
+	case reflect.Array, reflect.Slice:
+		return fmt.Sprintf("List of %s", toTypeDescription(t.Elem()))
+	case reflect.Ptr:
+		return fmt.Sprintf("Reference to %s", toTypeDescription(t.Elem()))
+	case reflect.Chan:
+		return fmt.Sprintf("Channel of %s", toTypeDescription(t.Elem()))
+	case reflect.Map:
+		return fmt.Sprintf("Map of %s to %s", toTypeDescription(t.Key()), toTypeDescription(t.Elem()))
+	case reflect.Func:
+		return "Function"
+	case reflect.Interface:
+		return "Interface"
+	case reflect.Struct:
+		if t.Name() != "" {
+			return t.Name()
+		}
+		return "Nested Struct"
+	case reflect.String:
+		name := t.Name()
+		if name != "" && name != "string" {
+			return name
+		}
+		return "String"
+	case reflect.Bool:
+		name := t.Name()
+		if name != "" && name != "bool" {
+			return name
+		}
+		return "Boolean"
+	case reflect.Int:
+		name := t.Name()
+		if name != "" && name != "int" {
+			return name
+		}
+		return "Integer"
+	case reflect.Int8:
+		name := t.Name()
+		if name != "" && name != "int8" {
+			return name
+		}
+		return "Integer(8 bits)"
+	case reflect.Int16:
+		name := t.Name()
+		if name != "" && name != "int16" {
+			return name
+		}
+		return "Integer(16 bits)"
+	case reflect.Int32:
+		name := t.Name()
+		if name != "" && name != "int32" {
+			return name
+		}
+		return "Integer(32 bits}"
+	case reflect.Int64:
+		name := t.Name()
+		if name != "" && name != "int64" {
+			return name
+		}
+		return "Integer(64 bits)"
+	case reflect.Uint:
+		name := t.Name()
+		if name != "" && name != "uint" {
+			return name
+		}
+		return "Unsigned Integer"
+	case reflect.Uint8:
+		name := t.Name()
+		if name != "" && name != "uint8" {
+			return name
+		}
+		return "Unsigned Integer(8 bits)"
+	case reflect.Uint16:
+		name := t.Name()
+		if name != "" && name != "uint16" {
+			return name
+		}
+		return "Unsigned Integer(16 bits)"
+	case reflect.Uint32:
+		name := t.Name()
+		if name != "" && name != "uint32" {
+			return name
+		}
+		return "Unsigned Integer(32 bits}"
+	case reflect.Uint64:
+		name := t.Name()
+		if name != "" && name != "uint64" {
+			return name
+		}
+		return "Unsigned Integer(64 bits)"
+	case reflect.Float32:
+		name := t.Name()
+		if name != "" && name != "float32" {
+			return name
+		}
+		return "Float"
+	case reflect.Float64:
+		name := t.Name()
+		if name != "" && name != "float64" {
+			return name
+		}
+		return "Float(64 bits)"
+	case reflect.Complex64:
+		return "Complex(64 bits)"
+	case reflect.Complex128:
+		return "Complex(128 bits)"
+	}
+	return fmt.Sprintf("%+v", t)
+}
+
+// Usage writes usage information to stderr using the default header and table format
+func Usage(prefix string, spec interface{}) error {
+	return Usagef(prefix, spec, os.Stderr, DefaultHeader, DefaultTableFormat)
+}
+
+// Usagef writes usage information to the specified io.Writer using the specifed header and usage format
+func Usagef(prefix string, spec interface{}, out io.Writer, header string, format string) error {
```

Comment:
We don't need to provide so much structure, the templating language can handle it better. If we just pass a single user-provided template all the usageInfos at once in a `[]usageInfo`, they could accomplish this with a template like:

```
header here
{{ range . }}
per-info item here
{{ end }}
```